server/controllers/AuthorController.js

```diff
@@ -29,4 +29,50 @@ export default class AuthorController {
       return responseMessage(response, 500, { message: error.message });
     }
   }
+
+  /**
+   * @description allows admin to update an author
+   * @param {object} request express request object
+   * @param {object} response express response object
+   * @returns {json} json
+   * @memberof AuthorControllers
+   */
+  static async updateAuthor(request, response) {
+    const { params: { authorId }, body: { authorName } } = request;
+    try {
+      const existingAuthor = await Author.findOne({ where: { id: authorId } });
+      if (!existingAuthor) { return responseMessage(response, 404, { message: 'author not found' }); }
+      const sameAuthor = await Author.findOne({ where: { fullname: authorName } });
```

Comment:
This const existingAuthor = await Author.findOne({ where: { id: authorId } }); and
const sameAuthor = await Author.findOne({ where: { fullname: authorName } }); are thesame,
why not get all author with  Author.findAll(), and then use map and find to get the author you want, insteading constantly querying the database.